pkg/workflow/transitioner.go

```diff
@@ -0,0 +1,332 @@
+package workflow
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/nov1n/kubernetes-workflow/pkg/api"
+	"github.com/nov1n/kubernetes-workflow/pkg/client"
+	"github.com/nov1n/kubernetes-workflow/pkg/client/cache"
+	"github.com/nov1n/kubernetes-workflow/pkg/controller"
+	"github.com/nov1n/kubernetes-workflow/pkg/validation"
+	k8sApi "k8s.io/kubernetes/pkg/api"
+	k8sApiErr "k8s.io/kubernetes/pkg/api/errors"
+	k8sApiUnv "k8s.io/kubernetes/pkg/api/unversioned"
+	k8sClSet "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	k8sClSetUnv "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/unversioned"
+	k8sRec "k8s.io/kubernetes/pkg/client/record"
+	k8sUtRunt "k8s.io/kubernetes/pkg/util/runtime"
+)
+
+const (
+	workflowUIDLabel               = "workflow-uid"
+	workflowValidLabel             = "valid"
+	requeueAfterStatusConflictTime = 500 * time.Millisecond
+	requeueJobstoreNotSyncedTime   = 100 * time.Millisecond
+	retryOnStatusConflict          = 3
+	falseString                    = "false"
+)
+
+// Transitioner is responsible for transitioning a workflow from its current
+// state to a desired state.
```

Comment:
to -> towards? As it is not guaranteed that we can immediately move to the desired state